ui/lib/pki/addon/decorators/check-config.js

```diff
@@ -0,0 +1,36 @@
+import Route from '@ember/routing/route';
+
+/**
+ * the overview, roles, issuers, certificates, and key routes all need to be aware of the whether there is a config for the engine
+ * if the user has not configured they are prompted to do so in each of the routes
+ * decorate the necessary routes to perform the check in the beforeModel hook since that may change what is returned for the model
+ */
+
+export function withConfig() {
```

Comment:
It looks like the implementation is quite similar to the `fetch-config` decorator in the Kubernetes addon. I wasn't sure when I created it if this would be a reoccurring pattern or not but it looks like there are enough similarities that it's worth considering a more general approach.

What do you think about moving this to the core addon and updating it to accept arguments for `model` and `queryOptions`? The `configModel` and `configError` properties could be added from the k8s implementation and they could be accessed on an as needed basis.

We could go as far as not relating it to config at all since you could query for anything that you might need to prompt for, so you could change some of the verbiage for even wider usage.

Reply:
I think it's fine keeping these separate for now given the challenges around passing in query options that depend on a class property though.